### ISSUE TITLE
3.1.0 release

### ISSUE THREAD

**Issue by blairconrad** [FakeItEasy/FakeItEasy#986]
Can be labelled **ready** when all other issues on this milestone are closed.

- [x] run code analysis in VS in *Release* mode and address violations (send a regular PR which must be merged before continuing)
- [x] if necessary, change `VERSION_SUFFIX` in [AppVeyor](https://ci.appveyor.com/project/FakeItEasy/fakeiteasy/settings) to  "" and initiate a build
- [x] check build
- edit draft release in [GitHub UI](https://github.com/FakeItEasy/FakeItEasy/releases):
    - [x] ensure completeness of release notes, including non-owner contributors, if any (move release notes forward from any pre-releases to the current release)
    - [x] attach main nupkg and/or analyzer nupkg, whichever have content to release
    - [x] publish the release
- [x] push nupkgs to NuGet using the Deploy option on the AppVeyor build
- [x] de-list pre-release or superseded buggy NuGet packages if present
- [x] tweet, mentioning contributors and post link as comment here for easy retweeting ;-)
- [x] post tweet in [Gitter](https://gitter.im/FakeItEasy/FakeItEasy)
- [x] post a link to the GitHub Release in each issue in this milestone, with thanks to contributors
- ~~run `rake next_version[new_version]` to~~
    - [x] create a pull request that changes the version in CommonAssemblyInfo.cs to the expected version (of form _xx.yy.zz_) (#1014)
    - [x] create a new draft GitHub Release
    - [ ] create a new milestone for the next release
    - [x] create a new issue (like this one) for the next release, adding it to the new milestone
- [x] close this milestone

**Comment by blairconrad** [comment]
@adamralph, @thomaslevesque and I were chatting and think we have enough to release (and it's been ages since 3.0.0). Objections to a release tomorrow, if we can get our act in gear?

**Comment by adamralph** [comment]
No objection at all! :shipit: 

Indeed, in FakeItEasy terms, it has been ages!

**Comment by blairconrad** [comment]
https://twitter.com/fakeiteasyfx/status/840155389052256256
